tests/integ/test_dataframe.py

```diff
@@ -2437,10 +2437,22 @@ def test_describe(session):
     assert "invalid identifier" in str(ex_info)
 
 
+
+
+def test_truncate_existing_table(
+    session
+):
+    table_name = Utils.random_name_for_temp_object(TempObjectType.TABLE)
+    df = session.create_dataframe([(1, 2), (3, 4)]).toDF("a", "b")
+    df.write.save_as_table(table_name, mode="overwrite", table_type="temp")
```

Comment:
can we add a test that ensures that truncate is working as expected. For example a test that demonstrates table schema stays intact when using `truncate` whereas it is overwritten when using `overwrite`.